api/src/main/java/io/grpc/GlobalInterceptors.java

```diff
@@ -0,0 +1,144 @@
+  private boolean isGlobalInterceptorsTracersSet;
+  private boolean isGlobalInterceptorsTracersGet;
+
+  @VisibleForTesting
+  GlobalInterceptors() {}
+
+  /** Returns the GlobalInterceptors instance. */
+  private static synchronized GlobalInterceptors getInstance() {
+    if (instance == null) {
+      instance = new GlobalInterceptors();
+    }
+    return instance;
+  }
+
+  /**
+   * Sets the list of global interceptors and global server stream tracers.
+   *
+   * <p>If {@code setInterceptorsTracers()} is called again, this method will throw {@link
+   * IllegalStateException}.
+   *
+   * <p>It is only safe to call early. This method throws {@link IllegalStateException} after any of
+   * the get calls [{@link #getClientInterceptors()}, {@link #getServerInterceptors()} or {@link
+   * #getServerStreamTracerFactories()}] has been called, in order to limit changes to the result of
+   * {@code setInterceptorsTracers()}.
+   *
+   * @param clientInterceptorList list of {@link ClientInterceptor} that make up global Client
+   *     Interceptors.
+   * @param serverInterceptorList list of {@link ServerInterceptor} that make up global Server
+   *     Interceptors.
+   * @param serverStreamTracerFactoryList list of {@link ServerStreamTracer.Factory} that make up
+   *     global ServerStreamTracer factories.
+   */
+  public static void setInterceptorsTracers(
+      List<ClientInterceptor> clientInterceptorList,
+      List<ServerInterceptor> serverInterceptorList,
+      List<ServerStreamTracer.Factory> serverStreamTracerFactoryList) {
+    try {
+      GlobalInterceptors.getInstance()
+          .setGlobalInterceptorsTracers(
+              clientInterceptorList, serverInterceptorList, serverStreamTracerFactoryList);
+    } catch (Exception exception) {
+      throw exception;
+    }
+  }
+
+  /** Returns the list of global {@link ClientInterceptor}. If not set, this returns null. */
+  public static List<ClientInterceptor> getClientInterceptors() {
+    return GlobalInterceptors.getInstance().getGlobalClientInterceptors();
+  }
+
+  /** Returns list of global {@link ServerInterceptor}. If not set, this returns null. */
+  public static List<ServerInterceptor> getServerInterceptors() {
+    return GlobalInterceptors.getInstance().getGlobalServerInterceptors();
+  }
+
+  /** Returns list of global {@link ServerStreamTracer.Factory}. If not set, this returns null. */
+  public static List<ServerStreamTracer.Factory> getServerStreamTracerFactories() {
+    return GlobalInterceptors.getInstance().getGlobalServerStreamTracerFactories();
+  }
+
+  @VisibleForTesting
+  synchronized void setGlobalInterceptorsTracers(
+      List<ClientInterceptor> clientInterceptorList,
+      List<ServerInterceptor> serverInterceptorList,
+      List<ServerStreamTracer.Factory> serverStreamTracerFactoryList) {
+    if (isGlobalInterceptorsTracersGet) {
+      throw new IllegalStateException("Set cannot be called after any corresponding get call");
```

Comment:
nit: remove "corresponding" ? There is one setter and 3 different getters so I don't see what is "corresponding" to what

Reply:
Done